### PR TITLE
Add implementation details for find

### PR DESCRIPTION
**Pull request by w-yuchen**
Sorry for doing this so late, can just leave out if it's too poorly written.

### CODE REVIEW

**Review by w2vgd — DISMISSED** [review]
LGTM

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -342,6 +342,67 @@ The following sequence diagram shows how the update operation works:
 <div markdown="span" class="alert alert-info">:information_source: **Note:** The lifeline for `UpdateCommandParser` and `UpdateNewCommand` should end at the destroy marker (X) but due to a limitation of PlantUML, the lifeline reaches the end of diagram.
 </div>
 
+### 3.5 Find feature
+
+#### 3.5.1 Current Implementation
+
+The find mechanism is facilitated by `PocketEstate`. It implements the search feature that allows the user to specific the fields to search for and use multiple parameters.
```
Comment (w2vgd) [inline docs/DeveloperGuide.md]: `to specify` instead of `to specific`

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -342,6 +342,67 @@ The following sequence diagram shows how the update operation works:
...
 
+### 3.5 Find feature
+
+#### 3.5.1 Current Implementation
+
+The find mechanism is facilitated by `PocketEstate`. It implements the search feature that allows the user to specific the fields to search for and use multiple parameters. 
+This feature is implemented separately for both property and appointment, as well as an additional `find client` feature that searches both properties and appointments. 
+
+`find appointment` and `find property` uses `AppointmentPredicateList` and `PropertyPredicateList` to keep track of all 
+predicates that `FindAppointmentCommandParser` and `FindPropertyCommandParser` generated respectively. An unique predicate
```
Comment (w2vgd) [inline docs/DeveloperGuide.md]: `A unique`

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -342,6 +342,67 @@ The following sequence diagram shows how the update operation works:
...
+
+`find appointment` and `find property` uses `AppointmentPredicateList` and `PropertyPredicateList` to keep track of all 
+predicates that `FindAppointmentCommandParser` and `FindPropertyCommandParser` generated respectively. An unique predicate 
+class exist for every attribute that `Appointment` and `Property` classes have. `find client` command is implemented by 
+combining an `AppointmentNamePredicate` and a `PropertyClientNamePredicate`, and using the same keywords for both. 
+
+Since they are implemented similarly, given below is an example usage scenario and how the find mechanism behaves at each step for `find appointment`.  
+
+Step 1. The user launches the application for the first time. `PocketEstate` will be initialized with the initial 
+appointment book state and property book state, and the currentAppointmentBookStatePointer currentPropertyBookStatePointer pointing to the two initial book states respectively.
```
Comment (w2vgd) [inline docs/DeveloperGuide.md]: Is this supposed to be here?

**Review by w2vgd — COMMENTED** [review]
Some typos

File: docs/DeveloperGuide.md (hunk 3 above)
Comment (w-yuchen) [reply docs/DeveloperGuide.md]: i think that was a relic from copying part of an earlier section, removed.

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (w-yuchen) [reply docs/DeveloperGuide.md]: corrected

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (w-yuchen) [reply docs/DeveloperGuide.md]: corrected

**Review by w2vgd — APPROVED** [review]
LGTM

**Pull request merged**